src/python/ensembl/pytest_unittest.py

```diff
@@ -0,0 +1,173 @@
+    if server_url.password and server_url.password.startswith('$'):
+        server_url.password = os.environ[server_url.password[1:]]
+        config.option.server = str(server_url)
+    # Add global variables
+    pytest.dbs_dir = Path(__file__).parents[2] / 'tests' / 'databases'
+
+
+def pytest_make_parametrize_id(val: Any) -> str:
+    """Returns a readable string representation of `val` that will be used by @pytest.mark.parametrize calls.
+
+    `Pytest collection hook
+    <https://docs.pytest.org/en/latest/reference.html#_pytest.hookspec.pytest_make_parametrize_id>`_.
+
+    Args:
+        val: The parametrized value.
+
+    """
+    if isinstance(val, ExitStack):
+        return 'No error'
+    if isinstance(val, RaisesContext):
+        return val.expected_exception.__name__
+    return str(val)
+
+
+@pytest.fixture(name='db_factory', scope='session')
+def db_factory_(request: FixtureRequest) -> Generator:
+    """Yields a unit test database (:class:`UnitTestDB`) factory.
+
+    Args:
+        request: Access to the requesting test context.
+
+    """
+    created = {}  # type: Dict[str, UnitTestDB]
+    server_url = request.config.getoption('server')
+    def db_factory(src: os.PathLike, name: Optional[str] = None) -> UnitTestDB:
+        """Returns a unit test database (:class:`UnitTestDB`) object.
+
+        Args:
+            src: Directory path where the test database schema and content files are located. If a relative
+                path is provided, the root folder will be ``src/python/tests/databases``.
+            name: Name to give to the new database. See :meth:`UnitTestDB.__init__()` for more information.
+
+        """
+        src_path = Path(src) if os.path.isabs(src) else pytest.dbs_dir / src
+        db_key = name if name else src_path.name
+        return created.setdefault(db_key, UnitTestDB(server_url, src_path, name))
+    yield db_factory
+    # Drop all unit test databases unless the user has requested to keep them
+    if not request.config.getoption('keep_data'):
+        for test_db in created.values():
+            test_db.drop()
+
+
+@pytest.fixture(scope='session')
+def database(request: FixtureRequest, db_factory: Callable) -> Generator:
+    """Returns a unit test database (:class:`UnitTestDB`) object.
+
+    Requires a dictionary with keys `src` (mandatory) and `name` (optional) passed via `request.param`. See
+    :meth:`db_factory()` for details about each key's value. This fixture is a wrapper of :meth:`db_factory()`
+    intended to be used via indirect parametrization, for example::
+
+        @pytest.mark.parametrize("database", [{'src': 'master'}], indirect=True)
+        def test_method(..., database: UnitTestDB, ...):
+
+    Args:
+        request: Access to the requesting test context.
+
+    """
+    return db_factory(request.param['src'], request.param.get('name', None))
+
+
+@pytest.fixture(scope='session')
+def multi_dbs(request: FixtureRequest, db_factory: Callable) -> Dict:
```

Comment:
I find this name misleading because in the Perl API, "Multi" is an alias for "Compara" ...

As I understand, this is an extension of `database` above, but this one allows more than one databases. They should be named consistently (name shortening, etc), e.g. `db` and `dbs` or `multi_dbs`, or `database` and `databases` or `multiple_databases`

Reply:
Do we really care about PERL api? :-) Multi is as well an alias for ontology, taxonomy, etc... btw